src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -301,16 +316,26 @@ def add_two_photon_series(
     two_photon_series_metadata.update(imaging_plane=imaging_plane)
 
     # Add the data
-    def data_generator(imaging):
-        for i in range(imaging.get_num_frames()):
-            yield imaging.get_frames(frame_idxs=[i]).squeeze().T
-
-    data = H5DataIO(
-        data=DataChunkIterator(data_generator(imaging), buffer_size=buffer_size),
-        compression=True,
-    )
     two_p_series_kwargs = two_photon_series_metadata
-    two_p_series_kwargs.update(data=data)
+    if iterator_type == "v2":
+        data = H5DataIO(
+            data=ImagingExtractorDataChunkIterator(imaging_extractor=imaging, **iterator_options),
+            compression=True,
+        )
+        two_p_series_kwargs.update(data=data)
+    else:
+
+        def data_generator(imaging):
+            for i in range(imaging.get_num_frames()):
+                yield imaging.get_frames(frame_idxs=[i]).squeeze().T
+
+        if "buffer_size" not in iterator_options:
+            iterator_options.update(buffer_size=10)
+        data = H5DataIO(
+            data=DataChunkIterator(data_generator(imaging), **iterator_options),
+            compression=True,
+        )
+        two_p_series_kwargs.update(data=data)
```

Comment:
We should wrap this in an aux function.
Adding the details of which iterator is wrapped and how the parameters are propagated are details that do not concern the high level user of the `add_two_photon_series` (as the only things that matters is that the data is wrapped in an `h5dataio` iterator). Those details can easily be abstracted away to make this more readable.

Reply:
IDK how to do that - can you make a splinter PR pointing to this one as the base to do what you're imagining?

Reply:
I mean something like what we did for the ecephys pipeline:

https://github.com/catalystneuro/neuroconv/blob/27357f8145746f2279e580b8a5565eaa26a98f96/src/neuroconv/tools/spikeinterface/spikeinterface.py#L491-L550

Reply:
thanks @h-mayorquin for sharing this, I tried to do something similar here: https://github.com/catalystneuro/neuroconv/pull/91